File: równanie.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  
#  
def main(args):
    a = int(input("podaj liczbę a: "))
    b = int(input("podaj liczbę b: "))
    
    if a == 0: 
        if b == 0:
            print("nieskończenie wiele rozwiązań")
        else:
            print("równanie sprzeczne")
    else:
        x = -b / a
        print(x)
        
        


   
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: c++/równanie.cpp



#include <iostream>

using namespace std;

int main(int argc, char **argv) {
    int a;
    int b;
    float x;
	cout << "Witaj w C++";
   
    cout << "podaj liczbę a: ";
    cin >> a;	
    cout << "podaj liczbę b: ";
    cin >> b;
    if (a == 0) {
        if (b == 0) {
            cout << "nieskończenie wiele rozwiązań";
            } else {
                cout << "równaie sprzeczne"; 
            }
    } else {
        x = -b / (float)a;
        cout << x;
    }
	return 0;
}

